src/libraries/System.Runtime/tests/System.Runtime.Tests/System/ArrayTests.cs

```diff
@@ -1666,6 +1667,31 @@ public static void ConstrainedCopy_UnreliableConversion_ThrowsArrayTypeMismatchE
             Assert.Throws<ArrayTypeMismatchException>(() => Array.ConstrainedCopy(sourceArray, sourceIndex, destinationArray, destinationIndex, length));
         }
 
+        [Fact]
+        public static unsafe void Copy_PointerArray()
+        {
+            // Array of pointers is not supported by xUnit and test data properties with yield return
+
+            // Can copy between compatible pointers
+            uint*[] uintPointerArray = new uint*[1];
+            Array.ConstrainedCopy(new int*[1] { (int*)0x12345678 }, 0, uintPointerArray, 0, 1);
+            Assert.Equal((UIntPtr)0x12345678, (UIntPtr)uintPointerArray[0]);
```

Comment:
Add a test for incompatible pointers? (E.g. `bool*` and `int*`)

Reply:
There is an existing test at https://github.com/dotnet/runtime/blob/74dd85398a7f409cb3852cfc6cd3bd309e6fb2eb/src/libraries/System.Runtime/tests/System.Runtime.Tests/System/ArrayTests.cs#L1601-L1608

It should probably be renamed to "incompatible pointers".

Reply:
This test is between `void*[]` and `object[] (pointer array and object array)`. I meant to add one between `bool*[]` and `int*[]` (two incompatible pointer arrays).